Sources/ComposableArchitecture/Store.swift

```diff
@@ -362,15 +362,30 @@ public final class Store<State, Action> {
     self.scope(state: toLocalState, action: { $0 })
   }
 
-  func send(_ action: Action, originatingFrom originatingAction: Action? = nil) {
+  func send(_ action: Action, originatingFrom originatingAction: Action? = nil, instrumentation: Instrumentation = .shared) {
     self.threadCheck(status: .send(action, originatingAction: originatingAction))
 
+    let eventInfo = Instrumentation.EventInfo(
+      type: String(describing: Store.self),
+      action: debugCaseOutput(action),
+      tags: originatingAction.map { ["originating": debugCaseOutput($0)] } ?? [:]
+    )
+
     self.bufferedActions.append(action)
     guard !self.isSending else { return }
 
     self.isSending = true
     var currentState = self.state.value
+
+    instrumentation.store?.willSend(eventInfo)
+    defer {
+      instrumentation.store?.didSend(eventInfo)
+    }
+
     defer {
+      instrumentation.store?.willScope(eventInfo)
```

Comment:
nit: not super happy with the `scope` here. I think we're overloading this a bit. What about `updateState`?